CDCgov/prime-reportstream
# Update FHIRRouter Azure Events to Store messageID

This PR adds bundle.identifier to the Azure Custom Events logging. 

Test Steps:
1. Setting local Azure events logging, see [here](https://github.com/CDCgov/prime-reportstream/blob/platform/josh/14332-FHIRRouter-store-messageID/prime-router/docs/observability/local-dev-setup.md) for steps.
2. Post a report that will make it through the pipeline. Easiest way may be through running the end to end test: `./gradlew testEnd2EndUP`
3. Log into Azure with your SU account, navigate to `pdhtest-appinsights`, and open up the logs section.
4. Run the query `customEvents | where name == "ReportRouteEvent" or name == "ReportAcceptedEvent"`
5. Ensure that logs from your computer are being forwarded by looking at the cloud_RoleInstance or city/state columns.
6. Verify that MessageId is being appended to results in customDimensions. 
  a. The MessageId can be specifically pulled into it's own column for easier viewing: `customEvents | where name == "ReportRouteEvent" or name == "ReportAcceptedEvent" | extend MessageId = parse_json(tostring(customDimensions.messageId))`

## Changes
- Added bundle.identifier as "MessageId" to ReportAcceptedEvent and ReportRouteEvent
- Updated two FHIR data files with invalid FHIR syntax (Ex: `"meta.lastUpdated":"2023-09-29T07:57:58.484-07:00","identifier.system":"https://reportstream.cdc.gov/prime-router","identifier.value":"371784"`)
- Updated docs in a few places

## Checklist

### Testing

### Process

## Linked Issues
- Fixes https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/14332
